### DISCUSSION TITLE
SSE4.2 and POPCNT - which Windows and Office versions are affected?

### DISCUSSION THREAD

**Discussion by Torinde** [LegacyUpdate/LegacyUpdate#252] [Questions & Help]
In 2018 Microsoft began using SSE2 and that affected Windows versions that initially didn't require it like Win7, Server 2008 and [even XP](https://github.com/LegacyUpdate/LegacyUpdate/issues/201).

Win11 24H2 will [use POPCNT and checks for SSE4.2](https://www.tomshardware.com/software/windows/microsoft-updates-windows-11-24h2-requirements-cpu-must-support-sse42-or-the-os-will-not-boot). Usually new versions gets released in October, so right by the time Win7 support finishes. 

Win10, Office 2016 (and up), Server 2008 (and up) remain supported well beyond that date. So, will POPCNT/SSE4.2 be required after October 2024 for updates to:
- [ ] Win10 x86
- [ ] Win10 x64
- [ ] Office 2016 (and up) x86
- [ ] Office 2016 (and up) x64
- [ ] Win2008 x86
- [ ] Win2008 (and up) x64

Related:
- #321

**Comment by kirb** [comment]
I think it’s unlikely to affect downlevel updates, but still something to keep an eye on for sure.

My hunch is that the change occurred because it was around when stronger Spectre/Meltdown mitigations were being rushed into the OS, making it a combination of hardening the platform using routines written with newer instructions, and possibly the pressure to release ASAP causing the new compiler flags to be applied to everything, being internally tested and released long before anyone with a Pentium II/III/pre-Athlon 64 system could stop them. Of course, by the time it’s out and the complaints come in, Microsoft looks at it and goes, “we’re not going to compromise on our CPU mitigations to support 17+ year old computers”. This is purely my speculation, though. Perhaps analysis like I mentioned [here](https://github.com/LegacyUpdate/LegacyUpdate/issues/201#issuecomment-1878509243) against critical binaries in Windows 10 like win32k.sys could confirm this.

Should be noted that Windows 8 in 2012 required NX, PAE, and SSE2, and 8.1 in 2013 required CMPXCHG16b, PrefetchW, and LAHF/SAHF. Those requirements didn’t affect downlevel updates at the time, they (at least SSE2, possibly others) only come into the picture for downlevel in 2018. That, and Windows 11 has very unique hardware requirements unlike what any Microsoft software has depended on before, and I don’t think they would compromise their ability to deliver ads to Windows 10 users by bricking Core 2-era hardware (not a joke). So I think we’re fine, but if it ever does happen, I’ll jump on it.

**Reply by Torinde** [reply]
The speculation sounds reasonable.

How I see the requirements: 
- XP needs CMPXCHG8B 
- Win8 needs also PAE (not so much NX/SSE2 although those are officially listed) 
- Win10, only the x64 edition, needs/lists also CMPXCHG16b, PrefetchW, LAHF/SAHF
- updates after certain point in 2018, regardless of version (e.g. even for XP and Office), needs also SSE2 (which wasn't originally listed as requirement for XP/2008/7) 
- Win11 x64 lists SSE4.1 as official requirement (most SSE4.1 chips support CMPXCHG16b, PrefetchW, LAHF/SAHF), but doesn't actually need/use it
- update in 2024 (Win11 24H2 x64) lists SSE4.2 as official requirement and needs POPCNT (which is supported by all SSE4.2 chips and some without SSE4.2)

As you said - let's keep an eye on the effect of earlier versions, if any.
